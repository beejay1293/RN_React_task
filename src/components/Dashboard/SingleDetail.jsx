import React from 'react';
import propTypes from 'prop-types';
import { Col, Card, CardTitle, CardText, Table } from 'reactstrap';

const SingleDetail = ({
  name, updated, pchange1, pchange2, pchange3, price, created, sym, rank, supply
}) => (
  <Col sm="4">
  <Card body className="text-center DetailsCard">
    <CardTitle>{name} ({sym})</CardTitle>
    <p>Percentage Changes</p>
    <Table hover>
        <thead>
          <tr>
            <th>1 Hour</th>
            <th>27 Hours</th>
            <th>7 days</th>

          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{pchange1}</td>
            <td>{pchange2}</td>
            <td>{pchange3}</td>
          </tr>

        </tbody>
      </Table>
      <CardTitle>Price: {price}</CardTitle>
      <CardTitle>Total Supply: {supply}</CardTitle>
      <CardTitle>cmc Rank: {rank}</CardTitle>

    <CardText><b>Added on</b> {created}.</CardText>
    <CardText><b>Updated on</b> {updated}.</CardText>

  </Card>
</Col>
);

SingleDetail.propTypes = {
  name: propTypes.string.isRequired,
  price: propTypes.number.isRequired,
  rank: propTypes.number.isRequired,
  supply: propTypes.number.isRequired,
  pchange1: propTypes.number.isRequired,
  pchange2: propTypes.number.isRequired,
  pchange3: propTypes.number.isRequired,  
};
export default SingleDetail;
