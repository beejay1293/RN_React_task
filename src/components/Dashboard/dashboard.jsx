import React, { Component } from 'react';
import Header from './header';
import Details from './details';




export class Dashboard extends Component {

  render() {

    return (
      <section className="main">
        <Header />
        <Details/>
      </section>
    );
  }
}


