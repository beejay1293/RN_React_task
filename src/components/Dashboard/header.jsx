import React from 'react';

const Header = () => (
  <React.Fragment>
  <header className="header__container">
    <div className="logo">
      <h1 className="header">
          <span>
            <em>COIN</em>
          </span>
          <b className="coin">marketcap</b>
      </h1>
    </div>
  </header>
  </React.Fragment>
);

export default Header;
