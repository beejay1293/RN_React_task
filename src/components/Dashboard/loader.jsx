import React from 'react'
import Loader from 'react-loader-spinner'

const Loaders = () => (
    <div className='Loader'>
    <Loader
    type="Circles"
    color="#ffaf30"
    height={100}
    width={100}
    /> 
    </div>  
)

export default Loaders;

