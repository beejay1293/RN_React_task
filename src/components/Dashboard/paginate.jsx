import React from 'react';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';

const Paginate = ({items, changePage, page}) => (
    <Pagination aria-label="Page navigation" size="sm" >
      <PaginationItem disabled id={1} onClick={changePage}>
          <PaginationLink first id={1} onClick={changePage}/>
      </PaginationItem>
       <PaginationItem disabled id="prev" onClick={changePage}>
          <PaginationLink previous  id="prev" onClick={changePage}/>
       </PaginationItem>
     
          { 
            [...Array(items)].map((e, i) => {
                return  <PaginationItem key={i} id={i + 1} onClick={changePage} className={page === i + 1 ? 'active' : ''}> 
                <PaginationLink id={i + 1}>
                  {i + 1}
                </PaginationLink>
               </PaginationItem>
              })
        }
        
        <PaginationItem disabled id="next" onClick={changePage}>
          <PaginationLink next  id="next" onClick={changePage}/>
        </PaginationItem>
        <PaginationItem disabled id={items} onClick={changePage}>
          <PaginationLink last id={items}/>
        </PaginationItem>
      </Pagination>
)

export default Paginate;