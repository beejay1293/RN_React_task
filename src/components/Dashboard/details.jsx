import React, { Component } from 'react';
import SingleDetail from './SingleDetail';
import  { connect } from 'react-redux';
import Moment from 'react-moment';
import { withRouter } from 'react-router-dom';
import Paginate from './paginate';
import { getAllCryptocurrencyDetails } from '../../stores/actions/cryptocurrency';
import Loaders from './loader'

export class Detail extends Component {
  state = {
    currentPage: 1,
  };

   componentDidMount() { 
     const { getDetails } = this.props
     getDetails()
  }

  handleChangePage(e) {
    const { id } = e.target
    const { currentPage } = this.state
    const { getDetails } = this.props

    if(id === 'next'){
      let next = parseInt(currentPage, 10) + 1
       if(next > 10){
         next = 1
       }
      getDetails(next)
      this.setState({
        currentPage: next
      })
    }else if (id === 'prev'){
      let prev = parseInt(currentPage, 10) - 1
      if(prev < 1){
       prev = 0
      }
      getDetails(prev)
      this.setState({
        currentPage: prev
      })
    } else {
      getDetails(id)
      this.setState({
        currentPage: id
      })
      
    }    
  }


  render() {
    const { details, loading } = this.props;
    const { currentPage } = this.state;
    return (
      <div className="main__body">
       {loading ? (<Loaders />) : (<ul className="details">
          {details.map(detail => (
            <SingleDetail
            key = {detail.id}
            name = {detail.name}
            sym = {detail.symbol}
            supply= {detail.total_supply}
            updated= {<Moment format="LLLL">
            {detail.quote.USD.last_updated}
               </Moment>}
            pchange1 = {detail.quote.USD.percent_change_1h}
            pchange2 = {detail.quote.USD.percent_change_24h}
            pchange3 = {detail.quote.USD.percent_change_7d}
            price= {detail.quote.USD.price}
            created={<Moment format="LLLL">
            {detail.quote.USD.date_added}
               </Moment>}
            rank = {detail.cmc_rank}
            />
          ))}
       </ul>)  }
        <div className="paginate-box">
        <Paginate items={10} changePage={(e) => this.handleChangePage(e)} page={parseInt(currentPage, 10)}/>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  details: state.cryptocurrency.details,
  loading: state.cryptocurrency.isLoading
});
const mapDispatchToProps = {
  getDetails: getAllCryptocurrencyDetails,
};


export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Detail))
