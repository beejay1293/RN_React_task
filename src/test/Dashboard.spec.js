import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from '../components/Dashboard/dashboard';

describe('test Dashboard component', () => {
  it('should render component', () => {
    const Dash = shallow(<Dashboard  />);
    expect(Dash).toMatchSnapshot();
  });
});
