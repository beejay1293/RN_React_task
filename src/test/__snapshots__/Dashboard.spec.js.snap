// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Dashboard component should render component 1`] = `
<section
  className="main"
>
  <Header />
  <Connect(withRouter(Detail)) />
</section>
`;
